extensions/storage/README.md

```diff
@@ -0,0 +1,78 @@
+# Object Storage Extension Specification
+
+- **Title: Object Storage**
+- **Identifier: storage**
+- **Field Name Prefix: -**
+- **Scope: Item, Assets**
+- **Extension [Maturity Classification](../README.md#extension-maturity): Proposal**
+- **Owner**: @davidraleigh
+
+This document explains the fields of the Object Storage Extension to a STAC Item and an Asset. This does not cover NFS solutions provided by PaaS cloud companies. Allows user to specify details related to cloud storage access and costs for assets and their STAC item.
+
+- [Example (Landsat 8)](examples/example-landsat8.json)
+- [JSON Schema](json-schema/schema.json)
+
+## Asset fields
+
+| Field Name  | Type   | Description |
+| ----------- | ------ | ----------- |
+| storage:cloud_platform        | string    | The [cloud provider](#providers) where data is stored |
+| storage:manager               | string    | The entity in charge of managing the data. |
+| storage:regions               | [string]  | The region(s) where the data is stored. Relevant to speed of access and inter region egress costs (as defined by PaaS provider) |
+| storage:bucket                | string    | The bucket for the asset, used along with object path |
+| storage:object_path           | string    | The object_path for the asset, used along with bucket |
+| storage:requester_pays        | bool      | Is the data requester pays or is it data manager/cloud provider pays |
+| storage:tier                  | string    | The title for the tier type (as defined by PaaS provider) |
+| storage:tier_duration         | integer   | Minimum storage duration required before additional fees |
+| storage:date_stored           | string    | Date and time the corresponding asset placed into the current storage tier (relevant for tier_duration > 0) |
+| storage:first_byte_latency    | string    | time unit (milliseconds, minutes or hours) for accessing first byte of data |
```

Comment:
```suggestion
| storage:first_byte_latency    | string    | magnitude for time for accessing first byte of data (e.g., `milliseconds`, `seconds`,  `minutes`, or `hours`) |
```

Reply:
Could we make that a definitive list of options? (i.e. not "e.g." but "one of")

- milliseconds
- seconds
- minutes
- hours
- days
- weeks
- months
?

Reply:
i think hours is probably the highest we need to go.

Reply:
We have some companies that store on tape, which can take days to get the data from. Not sure this is covered by this extension though. And also not sure "first_byte_latency"  is the best term then

Reply:
I'm not convinced of this field is needed at all.  Doesn't `tier` imply a certain latency for a given provider?  
As a provider how would you populate this field? Would you basically look up the latency for the tier you are storing your data and use that?  If so then I don't think it is needed.

If is is needed, than `latency` would seem to be sufficient.

Reply:
@matthewhanson tier isn't directly related. For Alibaba's "Cold Archive", there are different options for your speed to first byte depending on the fees you're willing to pay, 1 hour, 2 hours, etc. 

it's not exactly latency. it really is a delay of some period of time and then your latency to download is the normal access speed. Oracle and Microsoft describe it as "Time to first byte" and AWS uses "First Byte Latency".

Reply:
I see, that makes sense. I don't have a strong opinion here, but agree with @m-mohr that this should be an enumerated list of options.